pytorch_lightning/trainer/trainer.py

```diff
@@ -984,9 +984,6 @@ def _run(self, model: "pl.LightningModule") -> Optional[Union[_EVALUATE_OUTPUT,
         self.data_connector.prepare_data()
         self.callback_connector._attach_model_callbacks()
 
-        if self._ckpt_path and not self.accelerator.restore_checkpoint_after_pre_dispatch:
```

Comment:
this was here for a reason. check deepspeed vs. training type base:

deepspeed: restore_checkpoint_after_pre_dispatch = True
all others: restore_checkpoint_after_pre_dispatch = False

Reply:
hm requiring validate/test/predict to restore model weights before setup while fit restores after seems like an odd reliance, does anyone have context/insights on use cases that would rely on this gap? originally introduced in PR #8352